src/audio/crossover/crossover.c

```diff
@@ -667,16 +667,22 @@ static int crossover_copy(struct comp_dev *dev)
 	}
 
 	/* Find the number of frames to copy over */
+	//printf("crossover_copy: src frames=%u\n",
+	//       audio_stream_get_avail_frames(&source->stream));
 	for (i = 0; i < num_sinks; i++) {
 		if (!sinks[i])
 			continue;
 		buffer_lock(sinks[i], &flags);
 		avail = audio_stream_avail_frames(&source->stream,
 						  &sinks[i]->stream);
 		frames = MIN(frames, avail);
+		printf("crossover_copy: sink[%d] frames avail=%u free=%u\n",
+		       i, audio_stream_get_avail_frames(&sinks[i]->stream),
+		       audio_stream_get_free_frames(&sinks[i]->stream));
 		buffer_unlock(sinks[i], flags);
 	}
 
+	printf("crossover_copy: frames to be proc=%u\n", frames);
```

Comment:
https://paste.googleplex.com/5753664419397632
From output log, "frames" becomes 0 because free frames of sink[1] is still 0 after the first crossover_copy() call, which should be back to 96 as sink[0] did.
"frames"=0 means no new data will be processed by "cd->crossover_process()" (line#699) during this call, and it causes deadlock.